src/__Private/LinterCLIConfig.php

```diff
@@ -146,27 +156,33 @@ public function getConfigForFile(string $file_path): self::TFileConfig {
         continue;
       }
 
-      $linters = Vec\concat(
-        $linters,
-        $override['extraLinters'] ?? vec[],
-      );
-      $blacklist = Vec\concat(
-        $blacklist,
-        $override['disabledLinters'] ?? vec[],
-      );
-      $autofix_blacklist = Vec\concat(
-        $autofix_blacklist,
-        $override['disabledAutoFixes'] ?? vec[],
-      );
-      $no_autofixes = $no_autofixes ||
-        ($override['disableAllAutoFixes'] ?? false);
+      if ($override['disableAllLinters'] ?? false) {
+        $linters = $normalize(vec[]);
+        $autofix_blacklist = $normalize(vec[]);
+        $linters = $assert_types($linters);
+        $autofix_blacklist = $assert_types($autofix_blacklist);
+        return shape(
+          'linters' => $linters,
+          'autoFixBlacklist' => $autofix_blacklist,
+        );
+      } else {
```

Comment:
This doesn't need to be in an `else` block, since you return early in the case of disabling all linters.